core/Settings.js

```diff
@@ -121,13 +124,47 @@ export default class Settings {
      */
     setSortElements(fn) {
         if (typeof(fn) !== "function") throw new Error(`[Amaterasu - #setSortElements] ${fn} is not a valid function`)
-
         this.sortElements = fn
-        this._reloadWindow()
 
         return this
     }
 
+    /**
+     * - Sets the starting x and y value of the gui
+     * @param {Number} x 
+     * @param {Number} y 
+     * @returns this for method chaining
+     */
+    setPos(x, y) {
+        this.bgPos = { x, y }
+        this.bgSize = this.bgSize ?? {w: 100 - (x * 2), h: 100 - (x * 2)}
+        
+        return this
+    }
+
+    /**
+     * - Sets the width and height of the gui
+     * @param {Number} w width
+     * @param {Number} h height
+     * @returns this for method chaining
+     */
+    setSize(w, h) {
+        this.bgSize = { w, h }
+        this.bgPos = this.bgPos ?? {x: (100 - w) / 2, y: (100 - h) / 2}
+        
+        return this
+    }
+
+    /**
+     * Applies all changes to the gui, must be set at the end of the method chain
+     */
+    apply() {
+        this.handler.getWindow().clearChildren()
+        this._init()
+
+        if (this.changelogText) this.addChangelog(this.changelogText)
+    }
+
```

Comment:
as for this, can you make it so if you change size or position it'll only change that instead of changing both in the same method ?
also we can have defaults so you can just set pos/size above those
e.g
```js
this.bgPos = { x: (20).percent(), y: (20).percent() }

// imagine this is inside the #setPos method
this.bgPos.x = (x).percent() /// etc... with size as well
```

Reply:
i'd appreciate if you dm me in discord and we can workout these issues more easily
this would make it so you're not constantly making code that will basically be deleted